x-pack/plugins/cases/public/components/connectors/servicenow/validate_json.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ValidationFunc } from '@kbn/es-ui-shared-plugin/static/forms/hook_form_lib';
+import { isEmpty } from 'lodash';
+import * as i18n from './translations';
+
+const MAX_ADDITIONAL_FIELDS_LENGTH = 10;
+
+export const validateJSON = (...args: Parameters<ValidationFunc>): ReturnType<ValidationFunc> => {
+  const [{ value }] = args;
+
+  try {
+    if (typeof value === 'string' && !isEmpty(value)) {
+      const parsedOtherFields = JSON.parse(value);
+
+      if (Object.keys(parsedOtherFields).length > MAX_ADDITIONAL_FIELDS_LENGTH) {
```

Comment:
I was able to add more than 10 fields and did not get validation error. 
<img width="884" alt="image" src="https://github.com/user-attachments/assets/c4b30dce-996b-4237-b963-753381ee65ff">

Reply:
I must have broken something fixing tests, damn it 😬

Thanks for testing!

Reply:
Ok I thought I was going crazy here. You tested with duplicated keys, I think that is why.

An error will be thrown if there are more than X unique keys.

Reply:
Ahh makes sense. 🤦‍♀️  I think they just override duplicate keys with last value, so if service now's apis don't have that validation, we don't need to validate.